Plugin install fails on Fedora with "invalid user syslog:dokku"

## Description of problem

Plugin installation fails on Fedora Server 33 with the following error:

```
$ sudo dokku plugin:install https://github.com/dokku-community/dokku-git-rev.git --name dokku-git-rev
-----> Cloning plugin repo https://github.com/dokku-community/dokku-git-rev.git to /var/lib/dokku/plugins/available/dokku-git-rev
Cloning into 'dokku-git-rev'...
remote: Enumerating objects: 98, done.
remote: Total 98 (delta 0), reused 0 (delta 0), pack-reused 98
Receiving objects: 100% (98/98), 14.78 KiB | 7.39 MiB/s, done.
Resolving deltas: 100% (38/38), done.
-----> Plugin dokku-git-rev enabled
chown: invalid user: ‘syslog:dokku’
```

I traced the issue to [`plugins/20_events/install`](https://github.com/dokku/dokku/blob/be01d2f7ec9903b6d0a96b1a0693c23cb82cb4ad/plugins/20_events/install#L20) and was able to monkey patch it into working like so:

```diff
--- /var/lib/dokku/plugins/available/20_events/install~	2021-06-03 13:00:49.942560314 +0000
+++ /var/lib/dokku/plugins/available/20_events/install	2021-06-03 13:02:29.334638854 +0000
@@ -17,7 +17,7 @@
   # shellcheck disable=SC2174
   mkdir -m 775 -p "$DOKKU_LOGS_DIR"
   case "$DOKKU_DISTRO" in
-    arch | debian | centos | rhel)
+    arch | debian | centos | rhel | fedora)
       chgrp dokku "$DOKKU_LOGS_DIR"
       ;;
     *)
@@ -28,7 +28,7 @@
   if [[ ! -f "$DOKKU_EVENTS_LOGFILE" ]]; then
     touch "$DOKKU_EVENTS_LOGFILE"
     case "$DOKKU_DISTRO" in
-      arch | debian | centos | rhel)
+      arch | debian | centos | rhel | fedora)
         chgrp dokku "$DOKKU_EVENTS_LOGFILE"
         ;;
       *)
```

The modification above got me past my original issue. I _would_ submit a PR but I don't have the means or knowledge necessary to test this in dokku as a whole and don't want to assume this is the only thing that needs to change for Fedora support.

This might become more of an issue if folks start moving away from CentOS and (presumably) onto Fedora once CentOS 7 is EOL.

## Environment

- Fedora Server 33
- Dokku 0.21.4 (installed from RHEL 8 RPMs)
- Docker CE 20.10.7

--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -12,7 +12,7 @@ set -eo pipefail
 # That's good because it prevents our output overlapping with wget's.
 # It also means that we can't run a partially downloaded script.
 
-SUPPORTED_VERSIONS="Debian [9, 10], CentOS [7], Ubuntu [16.04, 18.04, 20.04]"
+SUPPORTED_VERSIONS="Debian [9, 10], CentOS [7], Fedora (partial) [33, 34], Ubuntu [16.04, 18.04, 20.04]"
 
 log-fail() {
   declare desc="log fail formatter"
@@ -131,7 +131,7 @@ install-dokku-from-package() {
     debian | ubuntu)
       install-dokku-from-deb-package "$@"
       ;;
-    centos | rhel)
+    centos | fedora | rhel)
       install-dokku-from-rpm-package "$@"
       ;;
     *)

--- a/plugins/20_events/install
+++ b/plugins/20_events/install
@@ -17,7 +17,7 @@ trigger-events-install() {
   # shellcheck disable=SC2174
   mkdir -m 775 -p "$DOKKU_LOGS_DIR"
   case "$DOKKU_DISTRO" in
-    arch | debian | centos | rhel)
+    arch | debian | centos | fedora | rhel)
       chgrp dokku "$DOKKU_LOGS_DIR"
       ;;
     *)
@@ -28,7 +28,7 @@ trigger-events-install() {
   if [[ ! -f "$DOKKU_EVENTS_LOGFILE" ]]; then
     touch "$DOKKU_EVENTS_LOGFILE"
     case "$DOKKU_DISTRO" in
-      arch | debian | centos | rhel)
+      arch | debian | centos | fedora | rhel)
         chgrp dokku "$DOKKU_EVENTS_LOGFILE"
         ;;
       *)

--- a/plugins/nginx-vhosts/install
+++ b/plugins/nginx-vhosts/install
@@ -45,7 +45,7 @@ trigger-nginx-vhosts-install() {
       echo "%dokku ALL=(ALL) NOPASSWD:/usr/bin/systemctl reload $NGINX_INIT_NAME, $NGINX_BIN -t, $NGINX_BIN -t -c *" >"$NGINX_SUDOERS_FILE"
       ;;
 
-    centos | rhel)
+    centos | fedora | rhel)
       echo "%dokku ALL=(ALL) NOPASSWD:/usr/bin/systemctl reload $NGINX_INIT_NAME, $NGINX_BIN -t, $NGINX_BIN -t -c *" >"$NGINX_SUDOERS_FILE"
       echo "Defaults:dokku !requiretty" >>"$NGINX_SUDOERS_FILE"
       ;;

--- a/plugins/nginx-vhosts/internal-functions
+++ b/plugins/nginx-vhosts/internal-functions
@@ -119,7 +119,7 @@ fn-nginx-vhosts-nginx-init-cmd() {
       sudo /sbin/service "$NGINX_INIT_NAME" "$CMD"
       ;;
 
-    arch | centos | rhel)
+    arch | centos | fedora | rhel)
       sudo /usr/bin/systemctl "$CMD" "$NGINX_INIT_NAME"
       ;;
   esac

